internal/cephfs/mounter/kernel.go

```diff
@@ -72,24 +72,25 @@ func (m *kernelMounter) mountKernel(
 		m.needsModprobe = false
 	}
 
+	fsID, err := volOptions.GetFSID()
```

Comment:
why can't we use existing `GetFSID()` instead?
```suggestion
	fsID, err := volOptions.GetConnection().GetFSID()
```

Reply:
`volOptions.GetConnection()` can return `nil`, so it is not really safe. A new `GetFSID()` would prevent any incorrect usage, now, and possibly in the future.

Reply:
Yeah, I think the checking is needed, `Connect` and `Destroy` do the checking too.
And I've squashed the commits.

Reply:
> volOptions.GetConnection() can return nil, so it is not really safe. A new GetFSID() would prevent any incorrect usage, now, and possibly in the future.

when volOptions struct is created, we make sure we have the connection set. There is no way `GetConnection()` could return nil.

IMO, `GetConnection()` should be handling the nil check, 
```
func (vo *VolumeOptions) GetConnection() (*util.ClusterConnection, error) {
	if vo.conn == nil {
		return nil, errors.New("cluster not connected yet")
	}
	return vo.conn, nil
}
```

 as we have similar usage at some places. _for example_ -  https://github.com/ceph/ceph-csi/blob/72b9d5ac54d71e0f034a18a38d410a2da329c5e0/internal/cephfs/nodeserver.go#L145-L153

Reply:
Sure, improving `GetConnection()` would be nice too. I don't think that needs to be done in this PR though.

Reply:
> Sure, improving `GetConnection()` would be nice too. I don't think that needs to be done in this PR though.

That can be done through issue #5129.